### PR TITLE
Fix halo implementation and tiling artefact

### PR DESCRIPTION
**Pull request by qin-yu**
# Fix halo implementation and tiling artefact

1. Fix halo implementation from mirror padding to actual groundtruth
2. Fix normalisation from patch-level intensity normalisation to image-level
3. I recommend batch norm for training models for wide applicability

See my fix to `pytorch-3dunet` https://github.com/wolny/pytorch-3dunet/pull/113 for more details. 

image

### CODE REVIEW

**Comment by qin-yu** [comment]
I just fixed the problem in workflow:

- #219

But it seems that I need to add and modify tests

**Comment by lorenzocerrone** [comment]
Hey @Qin very nice, I like the pydantic validation a lot.. should we also add it to the plantseg configs? 
I think it would be so much nicer than the buggy validation I made.

**Comment by qin-yu** [comment]
Hey @lorenzocerrone, I 

- fixed the wrong halo implementation,
- changed intensity normalisation during inference to be image-level instead of patch-level,
- fixed the misuse of branch name in conda build workfllow

Please have a look

**Review by lorenzocerrone — APPROVED** [review]
Looks good we can merge it :)

**Comment by qin-yu** [comment]
I improved the docstring based on Adrian's feedback: https://github.com/wolny/pytorch-3dunet/pull/113#discussion_r1562584570

Since I'm already improving the docstrings, I also fixed the arg types and made some simple improvement of code.

I'll merge this and work on the next.

**Comment by qin-yu** [comment]
Removing the unused argument in `get_test_augmentations()` introduced a bug. Then I found that I might also have overlooked the previous global normalisation. Going to check it later.
